### ISSUE TITLE
Create a `testing.anndata` module

### ISSUE THREAD

**Issue by flying-sheep** [scverse/anndata#646]
Things I learned designing some of those:

- There should be a general layer (`testing.anndata`) with functions for anyone to import and test framework specific ones (e.g. `testing.anndata.fixtures`/`testing.anndata.pytest` which imports pytest and can be used as a pytest plugin)
- In Pytest, one can specify plugins, but if they’re specified in a `conftest.py` importing `anndata`, it will raise a `PytestAssertRewriteWarning`. An alternative is specifiying them as `addopts` in the TOML config (see below)
- We’re using to few fixtures, adding a few more for common tasks is more elegant than calling `adata = gen_adata()` as first line in many tests

My plan would be:

1. move all test functions to `testing.anndata` (alternative: use a private module until we could decide to do it)
2. create a Pytest plugin (i.e. a module containing fixture definitions) as `testing.anndata.fixtures`
3. use the plugin in anndata’s own tests with `-p anndata.testing.fixtures`

In summary, we’d add this to `pyproject.toml`:

```toml
[tool.pytest.ini_options]
...
addopts = [
   '-ptesting.anndata.fixtures',  # to avoid PytestAssertRewriteWarning
]
```

This would be a more gradual attempt than #477

**Comment by ivirshup** [comment]
I'm all for moving some testing functions to `anndata.testing`. Probably `assert_equal` and `gen_adata`. Functions for generating indexes into anndata objects could be useful too. But we should be conservative on the public methods included here.

---------

What's the advantage of a plug-in here?

I don't like the "globally available fixtures", that occur with `conftest.py`. I think we can just use the `pythonpath` [configuration value to keep some utilities in the `tests`](https://github.com/pytest-dev/pytest/pull/9134).

---------

> We’re using too little fixtures, adding a few more for common tasks is more elegant than calling adata = gen_adata() as first line in many tests

How would you replace `gen_adata` with a fixture? It requires an argument to be passed. It is often used within fixtures though.
